version.go

```diff
@@ -25,7 +26,9 @@ type versionsByNumber []*Version
 func (s versionsByNumber) Len() int      { return len(s) }
 func (s versionsByNumber) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 func (s versionsByNumber) Less(i, j int) bool {
-	return s[i].Number < s[j].Number
```

Comment:
There's probably a lot of 1.6 & 1.7 still out there and I'm not sure that dependent tools like Terraform would be very keen on moving to 1.8+ already.

If the api doesn't return an int we probably should reach out to Fastly directly. I haven't noticed that but it might be on older versions of services.

If you're ok with changing this field's type it would indeed make more sense to me. It also sounded weird to me why `Number` was declared as a `string` in the sdk, but then I figured that proposing this change could have a much bigger impact on the other tools relying on this library, that's why I proposed this change.